src/client/signatures/canonical_json/src/ser.rs

```diff
@@ -0,0 +1,421 @@
+use log::debug;
+use serde::ser::Serialize;
+use serde_json::ser::{CharEscape, Formatter};
+use std::io::Write;
+use std::string::FromUtf8Error as Utf8Error;
+
+struct JSONFormatter {}
+
+#[derive(Debug)]
+pub enum CanonicalJSONError {}
+
+impl From<Utf8Error> for CanonicalJSONError {
+    fn from(err: Utf8Error) -> Self {
+        err.into()
+    }
+}
+
+impl From<serde_json::error::Error> for CanonicalJSONError {
+    fn from(err: serde_json::error::Error) -> Self {
+        err.into()
+    }
+}
+
+impl Formatter for JSONFormatter {
+    fn write_f64<W: ?Sized>(&mut self, writer: &mut W, value: f64) -> Result<(), std::io::Error>
+    where
+        W: Write,
+    {
+        format_number(writer, value, 10.0_f64.powf(21.0), 10.0_f64.powf(-6.0))?;
+        Ok(())
+    }
+
+    fn write_char_escape<W: ?Sized>(
+        &mut self,
+        writer: &mut W,
+        char_escape: CharEscape,
+    ) -> Result<(), std::io::Error>
+    where
+        W: Write,
+    {
+        match char_escape {
+            CharEscape::Quote => {
+                writer.write("\"".as_bytes())?;
+            }
+            CharEscape::ReverseSolidus => {
+                writer.write("\\\\".as_bytes())?;
+            }
+            CharEscape::LineFeed => {
+                writer.write("\n".as_bytes())?;
+            }
+            CharEscape::Tab => {
+                writer.write("\t".as_bytes())?;
+            }
+            CharEscape::CarriageReturn => {
+                writer.write("\r".as_bytes())?;
+            }
+            CharEscape::Backspace => {}
+            CharEscape::FormFeed => {}
+            CharEscape::Solidus => {}
+            CharEscape::AsciiControl(_number) => {}
```

Comment:
`\u00{_number}` ?

Reply:
added custom serde's ser implemenation for AsciiControl